### DISCUSSION TITLE
Welcome to browser-extension Discussions!

### DISCUSSION THREAD

**Discussion by bdougie** [open-sauced-craftwork/browser-extension#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ishaanbedi** [comment]
Hi there @bdougie 👋🏻
I am Ishaan, a 20-year-old developer. Looking forward to contributing to OpenSauced, during and beyond octernships period!

While playing around with OpenSauced and the API, I've successfully figured out the way to make the browser extension work using vite + crxjs and I've spent quite some time today learning more about them!

Can you please have a look at the recordings of the same and let me know if this is something that meets the project's requirements?
I've tried to keep the design of the button as close as to that of github's UI so that it looks much more blended and beautiful.

***For a user registered on OpenSauced:***

https://user-images.githubusercontent.com/39641326/223487646-00d06b9c-7279-40e8-8e95-2fc03bb454e4.mov

***For a user not registered on OpenSauced:***

https://user-images.githubusercontent.com/39641326/223488822-5ea6f44c-9a89-479c-8c93-9147d8156f2f.mov

Thanks & Regards
Ishaan Bedi
(@ishaanbedi)

**Reply by bdougie** [reply]
This looks good. Feel free to go above and beyond and add more features.
some suggestions, 
- make the button OpenSauced orange
- list interests selected

**Reply by bdougie** [reply]
Also reminder to open a PR for me to review when you are ready.

**Reply by ishaanbedi** [reply]
> This looks good. Feel free to go above and beyond and add more features. some suggestions,
> 
> * make the button OpenSauced orange
> * list interests selected

Hi @bdougie !
As per your suggestions, here's what I've modified!

<img width="366" alt="Screenshot 2023-03-08 at 7 48 16 AM" src="https://user-images.githubusercontent.com/39641326/223602546-bfe21ff1-6e36-40f5-8797-a44a43557a10.png">


> Also reminder to open a PR for me to review when you are ready.

Definitely! 
After working upon [these ideas](https://github.com/open-sauced-craftwork/browser-extension/discussions/5#discussioncomment-5235808), I'll submit a PR for you to review.

Thanks & Regards
Ishaan Bedi
(@ishaanbedi)

**Reply by ishaanbedi** [reply]
Hi @bdougie !
I've made some progress as discussed. 

Things I've worked upon: 
- Caching Data: As discussed, I've written the algorithm for storing the data in localstorage and validating it after 7 days (the time frame can be changed) for any changes and updating it. Example: for the `username` of a user as the key, the value stored is:
```
{
   "data":{
      "id":"39641326",
      "open_issues":"0",
      "is_private":false,
      "is_open_sauced_member":true,
      "created_at":"2023-03-07T13:45:40.507Z",
      "updated_at":"2023-03-07T13:48:03.503Z",
      "login":"ishaanbedi",
      "is_onboarded":true,
      "is_waitlisted":false,
      "role":10,
      "bio":"",
      "name":"Ishaan Bedi",
      "url":"https://www.ishaanbedi.in/",
      "twitter_username":"ishnbedi",
      "linkedin_url":"https://www.linkedin.com/in/ishaanbedi13/",
      "github_sponsors_url":"",
      "company":"",
      "location":"India",
      "display_local_time":false,
      "interests":"javascript,react",
      "display_email":false,
      "receive_collaboration":false,
      "timezone":"India Standard Time"
   },
   "preservedTimeStamp":1678266009116
}
```
  We can compare `preservedTimeStamp` with `Date.now()` before making the API call, making this work somewhat close to caching mechanism.
- For inviting a user to the community, I've created the logic that shows the `View on OpenSauced` button if user is registered on the platform, else if we can scrape the twitter/email etc. from the user's bio, we can invite the user by generating a tweet/email exclusively for them. If none of these two is found, nothing is displayed so as to prevent dead clicking. We can make the invitation purpose more valuable by analysing future updates to the platform.

https://user-images.githubusercontent.com/39641326/223672733-451dafc3-7168-4a67-a102-6bb029290c1c.mp4

Looking forward for your feedback and remarks before I proceed with a PR.

Thanks & Regards
Ishaan Bedi
(@ishaanbedi)

**Comment by whopiyush** [comment]
Hello @bdougie o/
just completed the initial task. Here is how it looks rn:

1. If a GH user **has an** account on OpenSacued:
	
	![image](https://user-images.githubusercontent.com/18086566/223509720-66d55e3a-e65c-4e9f-949e-223d0816e80c.png)

2. If a GH user **does not** have an account on OpenSauced:
	
	![image](https://user-images.githubusercontent.com/18086566/223510147-dea91f2f-1137-4158-9424-29ca58a13eee.png)

would love to hear about any improvements and suggestions by your side

**Reply by bdougie** [reply]
Commented on your PR. Consider some other things you could add to stand out and showcase your skill. https://github.com/open-sauced-craftwork/browser-extension/discussions/5

**Comment by nickytonline** [comment]
Hey everyone! Just wanted to take a second to say hello! 👋🏻 Loving all the discussion here and the work being done! 👏🏻

**Comment by AyushAgnihotri2025** [comment]
Hey mentors, I have applied for the [Open Sauced Craftwork Browser Extension](https://github.com/open-sauced-craftwork/open_sauced_craftwork_browser_extension-AyushAgnihotri2025) but in my repo no **Feedback Pull Request** is generated. So, can you please help me out ? 

Thanks

**Reply by bdougie** [reply]
You will need to generate a pull request using these steps [here](https://github.com/open-sauced-craftwork/open_sauced_craftwork_browser_extension-AyushAgnihotri2025#task-submission).

The changes you need to make on your branch are [here](https://github.com/open-sauced-craftwork/open_sauced_craftwork_browser_extension-AyushAgnihotri2025#task-instructions).

**Reply by AyushAgnihotri2025** [reply]
@bdougie Sir, as mentioned [here](https://github.com/open-sauced-craftwork/open_sauced_craftwork_browser_extension-AyushAgnihotri2025#task-submission:~:text=Using%20the%20auto%20generated%20Feedback%20Pull%20Request%20for%20review%20and%20submission) In my repo no such **Feedback Pull Request** is generated.

**Reply by bdougie** [reply]
> @bdougie Sir, as mentioned [here](https://github.com/open-sauced-craftwork/open_sauced_craftwork_browser_extension-AyushAgnihotri2025#task-submission:~:text=Using%20the%20auto%20generated%20Feedback%20Pull%20Request%20for%20review%20and%20submission) In my repo no such **Feedback Pull Request** is generated.

You missed step 1, and didn't create a new branch. I only see `main` in your repo. Your work should be done on a new branch.

**Comment by asrvd** [comment]
Hey @bdougie , ashish here - a 19 y/o web developer and a student from India, I have completed the initial task given, and also made a PR, can I get a review please (I have also added some suggestions I had in the PR)

https://user-images.githubusercontent.com/68690233/224337292-d6f61ec8-de21-40d5-a2fa-be792a8c7246.mp4

https://github.com/open-sauced-craftwork/open_sauced_craftwork_browser_extension-asrvd/pull/1

**Comment by dapoadedire** [comment]
Hello, @bdougie .
I'm Dapo Adedire from Nigeria.
I just finished the first assignment, and I'm ready for review.

https://github.com/open-sauced-craftwork/open_sauced_craftwork_browser_extension-dapoadedire/pull/2

Here are some demo images.

![image](https://user-images.githubusercontent.com/95668340/224368141-c15f1071-2ab1-4abe-8324-fdaf651bfbfd.png)

![image](https://user-images.githubusercontent.com/95668340/224368222-df5e89b0-7c93-4381-bf2a-0b7798f07b02.png)



Thank you.

**Comment by MitAbhay** [comment]
Hey @bdougie, I have completed the extension and ready for review and suggestions :)
https://github.com/open-sauced-craftwork/open_sauced_craftwork_browser_extension-MitAbhay

Demo images : 
1.  If the user does not have an account, we can redirect him to open-sauced site to join

![Screenshot from 2023-03-10 23-38-34](https://user-images.githubusercontent.com/76659113/224392090-b9034d9f-cf67-4d68-9395-1efdd9c703ca.png)

2.  If user have an account on open-sauced, we are redirecting him to his/her profile

![Screenshot from 2023-03-10 23-37-58](https://user-images.githubusercontent.com/76659113/224392082-d984644c-8e64-4b3b-ae0e-c35beb2f5ab9.png)

**Comment by AyushAgnihotri2025** [comment]
Hello, @bdougie, I'm Ayush.
I just finished the first assignment, kindly once review it.

https://github.com/open-sauced-craftwork/open_sauced_craftwork_browser_extension-AyushAgnihotri2025/pull/2

**Reply by AyushAgnihotri2025** [reply]
> Hello, @bdougie, I'm Ayush. I just finished the first assignment, kindly once review it.
> 
> [open-sauced-craftwork/open_sauced_craftwork_browser_extension-AyushAgnihotri2025#2](https://github.com/open-sauced-craftwork/open_sauced_craftwork_browser_extension-AyushAgnihotri2025/pull/2)

@bdougie, I have updated my pull kindly have a look.

**Comment by thekingslee** [comment]
Hello everyone, I am Kingsley. I am an engineering student in Nigeria. I look forward to making new friends and acquaintances with this engagement.

**Reply by nickytonline** [reply]
Welcome @nworiekingslee! 👋

**Reply by thekingslee** [reply]
Thank you 🤗

**Comment by Pratham-19** [comment]
Hello @bdougie , I'm Pratham 2nd year student from Manipal University Jaipur.
I have created a PR pls look into it.
open-sauced-craftwork/open_sauced_craftwork_browser_extension-Pratham-19#2
Thanks

**Reply by thekingslee** [reply]
Hello, nice to e-meet you 😁

**Reply by nickytonline** [reply]
Thanks for the PR submission @Pratham-19 and nice to meet you! We'll get to reviewing your PR, it just takes a bit of time as there are many submissions. 😅

**Comment by Inzamamdev** [comment]
Hello Developers I am happy to connect with such intelligent people worldwide. My name is Inzamam Shaikh an Engineering student from India 2nd year student from Lokmanya Tilak College of Engineering.

**Comment by Rishav1707** [comment]
https://user-images.githubusercontent.com/97666287/227719498-79b0e99f-faa5-4425-bd44-0a5e4c7ae59b.mp4

## Using auth token, the extensions popup look like this-
![Pop_up](https://user-images.githubusercontent.com/97666287/227719735-8e19e16d-6d91-4700-921d-88f71c83da38.png)

I have successfully completed my assignment and am ready for the review.
@bdougie and @nickytonline I have created a PR, kindly look into it and suggest to me any changes you want.

**Comment by bobbyy16** [comment]
hello @bdougie just completed the assignment
[Screencast from 25-03-23 07:58:56 PM IST.webm](https://user-images.githubusercontent.com/62874675/227724393-d55b5b06-c547-4bb8-9fde-0d0301f10a23.webm)
If the user has alreadt logged in to open-sauced-pizza website using github, you can just click the button to view his profile on openSauced website, if not that button redirects to login page of openSauced website, so that user can login with github. Although I've tried to send an invitation mail. But we don't have an access to user's email in github api.
